src/pages/data/[projectId].tsx

```diff
@@ -0,0 +1,477 @@
+import React, { useEffect, useState } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faChartSimple } from '@fortawesome/free-solid-svg-icons';
+import { faChartPie } from '@fortawesome/free-solid-svg-icons';
+import { faChartBar } from '@fortawesome/free-solid-svg-icons';
+import { faArrowRight } from '@fortawesome/free-solid-svg-icons';
+import dynamic from 'next/dynamic';
+
+import Label from '../../components/forms/inputs/label';
+import SelectInput from '../../components/forms/inputs/selectInput';
+import SquareField from '../../components/decoration/squares';
+import { trpc } from '../../lib/client/trpc';
+import prisma from '../../lib/db/prisma';
+import DateRangeSelectInput from '../../components/layout/data/dateRangeSelectInput';
+import useReactPath from '../../hooks/useReactPath';
+
+import type { DateRange } from '../../components/layout/data/dateRangeSelectInput';
+import type { OptionType } from '../../components/forms/inputs/selectInput';
+import type {
+  GetStaticPaths,
+  GetStaticProps,
+  InferGetStaticPropsType,
+} from 'next';
+import type {
+  DataDashboardProject,
+  DataDashboardData,
+  DataDashboardValue,
+} from '@prisma/client';
+import type { TimeSeriesData } from '../../components/layout/data/charts/timeSeriesLineChart';
+
+const TimeSeriesLineChart = dynamic(
+  () => import('../../components/layout/data/charts/timeSeriesLineChart'),
+  {
+    ssr: false,
+  }
+);
+
+/** Type representing a DataDashboardProject filled with data recursively filled with values. */
+type FilledDataDashboardProject = DataDashboardProject & {
+  data: (DataDashboardData & { values: DataDashboardValue[] })[];
+};
+
+/**
+ * Function that computes the lower bound of a date range according to a given "date range id".
+ * @param dateRange {DateRange | undefined} The id of a date range from which the lower date bound is computed.
+ * @return {Date | null} The lower date bound if it exist.
+ */
+const getDateRangeMinimum = (dateRange: DateRange | undefined): Date | null => {
+  switch (dateRange) {
+    case '7d':
+      return new Date(Date.now() - 7 * 24 * 60 * 60 * 1000);
+    case '30d':
+      return new Date(Date.now() - 30 * 24 * 60 * 60 * 1000);
+    case '60d':
+      return new Date(Date.now() - 60 * 24 * 60 * 60 * 1000);
+    case '90d':
+      return new Date(Date.now() - 90 * 24 * 60 * 60 * 1000);
+    case '6m':
+      const date = new Date();
+      date.setMonth(date.getMonth() - 6);
+      return date;
+    case 'YTD':
+      return new Date(new Date().getFullYear(), 0, 1);
+    case 'All time':
+      return null;
+    default:
+      return null;
+  }
+};
```

Comment:
Since the date-range is handled on a few different places I would recommend to create an array or object like this 
[{
range: '7d',
label: '7 days'
min: new Date(Date.now() - 7 * 24 * 60 * 60 * 1000);
...
}] or something like this to make it easier to change the ranges afterwards

Reply:
I created an associative dictionary that forces that every range in the union type `DateRange` is used. It contains useful attributes related to each date range. I don't know if its declaration in the module is correct or if it must be moved somewhere else. Any suggestion on this?

Reply:
Looking good :+1: 
I don't see any better place for it now, but moving it afterwards is easy anyways